runtime/mangata-rococo/src/lib.rs

```diff
@@ -133,17 +131,38 @@ pub type Executive = frame_executive::Executive<
 	AllPalletsWithSystem,
 >;
 
-use frame_support::traits::OnRuntimeUpgrade;
+pub struct MangataMigrations;
+
+impl OnRuntimeUpgrade for MangataMigrations {
+	fn on_runtime_upgrade() -> frame_support::weights::Weight {
+		migrations::asset_registry::AssetRegistryMigration::on_runtime_upgrade()
+	}
+
+	#[cfg(feature = "try-runtime")]
+	fn pre_upgrade() -> Result<(), &'static str> {
+		migrations::asset_registry::AssetRegistryMigration::pre_upgrade()
+			.expect("try-runtime pre_upgrade for AssetRegistryMigration failed!!");
+		Ok(())
+	}
+
+	#[cfg(feature = "try-runtime")]
+	fn post_upgrade() -> Result<(), &'static str> {
+		migrations::asset_registry::AssetRegistryMigration::post_upgrade()
+			.expect("try-runtime post_upgrade for AssetRegistryMigration failed!!");
+		Ok(())
+	}
+}
```

Comment:
actually it feels like there is a better way to register migrations by using `hook`. i think @iStrike7 did that for vesting ?